lib/traject/macros/mods.rb

```diff
@@ -16,9 +18,73 @@ def extract_name(role: nil, exclude: nil)
       extract_mods("/*/mods:name[mods:role/mods:roleTerm/#{clause}]/mods:namePart")
     end
 
+    def normalize_type
+      extract_mods('/*/mods:typeOfResource', translation_map: 'types')
+    end
+
+    def normalize_language
+      extract_mods('/*/mods:language/mods:languageTerm', translation_map: 'languages')
+    end
+
+    def normalize_script
+      extract_mods('/*/mods:language/mods:scriptTerm', translation_map: 'scripts')
+    end
+
+    def generate_has_part
+      lambda { |record, accumulator|
+        url = record.xpath('/*/mods:relatedItem[@type="constituent"]/mods:location/mods:url', NS).map(&:text)
+        title = record.xpath('/*/mods:relatedItem[@type="constituent"]/mods:titleInfo/mods:title', NS).map(&:text)
+
+        if url.present?
+          accumulator.concat(url)
+        elsif title.present?
+          accumulator.concat(title)
+        end
+      }
+    end
+
+    def generate_part_of
+      lambda { |record, accumulator|
+        url = record.xpath('/*/mods:relatedItem[@type="host"]/mods:location/mods:url', NS).map(&:text)
+        title = record.xpath('/*/mods:relatedItem[@type="host"]/mods:titleInfo/mods:title', NS).map(&:text)
+
+        if url.present?
+          accumulator.concat(url)
+        elsif title.present?
+          accumulator.concat(title)
+        end
+      }
+    end
+
+    def generate_series
```

Comment:
Maybe we could pull out a simple method that all these could share? I'm not sure it's worth writing individual tests for each variation, but between all 3 we might cover them with our integration test?